### ISSUE TITLE
Fix numeric type check for np.float32

### ISSUE THREAD

**Issue by AntoineGautier** [lbl-srg/funnel#41]
This is to address https://github.com/lbl-srg/BuildingsPy/issues/291.

`isinstance(np.float32(x), float)` returns `False`.

We coud replace
```
num_check = [isinstance(x, numbers.Real) for x in all_data]
```
With
```
def test_num(x):
    try:
        x - 0
        return True
    except TypeError:
        return False

num_check = [test_num(x) for x in all_data]
```

**Comment by Mathadon** [comment]
Another option would be to do a cast to `np.float64` at the point where the data is created?

**Comment by AntoineGautier** [comment]
Closed by #43 which specifies a minimum version for NumPy.
